packages/thycotic_ss/data_stream/logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,495 @@
+      ignore_missing: true
+
+  - grok:
+      field: cef.extensions.destinationUserName
+      patterns:
+        - '%{DATA:_tmp.destination_user_leading_domain}\\%{DATA:thycotic_ss.event.user.name}@%{GREEDYDATA:thycotic_ss.event.user.domain}'
+        - '%{DATA:thycotic_ss.event.user.name}@%{GREEDYDATA:thycotic_ss.event.user.domain}'
+        - '%{GREEDYDATA:thycotic_ss.event.user.domain}\\%{DATA:thycotic_ss.event.user.name}'
+        - '%{GREEDYDATA:thycotic_ss.event.user.name}'
+      ignore_missing: true
+      if: ctx.cef?.extensions?.destinationUserName != null && ctx.cef?.extensions?.destinationUserName != ""
+
+  - append:
+      field: related.user
+      value:
+        - '{{thycotic_ss.event.user.name}}'
+      if: ctx.thycotic_ss?.event?.user?.name != null && ctx.thycotic_ss?.event?.user?.name != ""
+
+  - remove:
+      field:
+        - thycotic_ss.event.user
+      ignore_missing: true
+      if: ctx.user?.name == ctx.thycotic_ss?.event?.user
+      description: "Remove duplication of user details under thycotic_ss.event.user if the same details are already in ECS user"
+
+  ##################################################################################################################################################
+  ## Thycotic SS event time, NOT stored as @timestamp as a significant delay often appears to exist between the event in SS and the syslog action ##
+  ##################################################################################################################################################
+
+  - date:
+      field: cef.extensions.deviceReceiptTime
+      target_field: thycotic_ss.event.time
+      formats:
+        - ISO8601
+      description: "Store CEF time as a thycotic_ss specific field, some delay appears to exist between event occurrence and Thycotic SS log transmission"
+
+  #############
+  ## message ##
+  #############
+
+  - convert:
+      field: cef.extensions.message
+      type: string
+      target_field: message
+      ignore_missing: true
+      description: "The CEF message should now be stored in message"
+
+  #####################################
+  ## Standard ECS Field Enrichments  ##
+  #####################################
+
+  - geoip:
+      field: source.ip
+      target_field: source.geo
+      ignore_missing: true
+
+  - geoip:
+      database_file: GeoLite2-ASN.mmdb
+      field: source.ip
+      target_field: source.as
+      properties:
+        - asn
+        - organization_name
+      ignore_missing: true
+
+  - rename:
+      field: source.as.asn
+      target_field: source.as.number
+      ignore_missing: true
+
+  - rename:
+      field: source.as.organization_name
+      target_field: source.as.organization.name
+      ignore_missing: true
+
+  ####################
+  ## Cleanup Fields ##
+  ####################
+
+  - remove:
+      field: event.original
+      if: "ctx?.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```

```suggestion
      if: "ctx.tags == null || !(ctx.tags.contains('preserve_original_event'))"
```
Conditional deref is not required on ctx since it is always present. Similar below.